src/SMAIAXBackend.Application/Services/Implementations/SmartMeterListService.cs

```diff
@@ -43,14 +44,15 @@ public async Task<SmartMeterOverviewDto> GetSmartMeterByIdAndUserIdAsync(Guid sm
             throw new SmartMeterNotFoundException(smartMeterId, validatedUserId.Id);
         }
 
-        var smartMeterOverviewDto = SmartMeterOverviewDtoFromSmartMeter(smartMeter);
+        var smartMeterOverviewDto = await SmartMeterOverviewDtoFromSmartMeter(smartMeter, validatedUserId);
 
         return smartMeterOverviewDto;
     }
 
-    private static SmartMeterOverviewDto SmartMeterOverviewDtoFromSmartMeter(SmartMeter smartMeter)
+    private async Task<SmartMeterOverviewDto> SmartMeterOverviewDtoFromSmartMeter(SmartMeter smartMeter, UserId userId)
```

Comment:
why not make this a method on SmartMeterOverviewDto? That way it could be used more universally